pageserver/src/tenant/storage_layer.rs

```diff
@@ -204,23 +204,52 @@ impl Default for ValuesReconstructState {
     }
 }
 
-/// Description of layer to be read - the layer map can turn
-/// this description into the actual layer.
-#[derive(PartialEq, Eq, Hash, Debug, Clone)]
-pub(crate) enum ReadableLayerDesc {
-    Persistent {
-        desc: PersistentLayerDesc,
-        lsn_range: Range<Lsn>,
-    },
-    InMemory {
-        handle: InMemoryLayerHandle,
-        lsn_ceil: Lsn,
-    },
+/// A key that uniquely identifies a layer in a timeline
+#[derive(Debug, PartialEq, Eq, Clone, Hash)]
+pub(crate) enum LayerKey {
+    PersitentLayerKey(PersistentLayerKey),
+    InMemoryLayerKey(InMemoryLayerKey),
```

Comment:
AGain, why not call it `LayerId` or stick with the `Desc`riptor convention? Key is confusing becaucse it already has a another meaning in the code base.

Reply:
Mostly because of the pre-existing `PersistentLayerKey` name. Switched to `Id` in https://github.com/neondatabase/neon/pull/7215/commits/0516d0e6de87c40588ca700bbe960322eb9ad747